src/log.rs

```diff
@@ -447,6 +549,10 @@ pub struct IndexLogOverlay {
 // We use identity hash for value overlay/log records so that writes to value tables are in order.
 #[derive(Debug, Default)]
 pub struct ValueLogOverlay {
+	pub map: HashMap<u64, (u64, Vec<u8>)>, // index -> (record_id, entry)
```

Comment:
Not too sure why we remove the BuildIdHash here

Reply:
The only reason for it is so that log changes are ordered by entry index when written to the disk. So it only makes sense to use in the local overlay. Ideally, local overlay should maintain a an additional sorted list for this. Identity hash for non-random indicies degrades in-memory performance.

Reply:
make sense, thanks for the explanation.